lib/counter.js

```diff
@@ -16,91 +16,92 @@ var extend = require('util-extend');
 
 var getLabels = require('./util').getLabels;
 
-/**
- * Counter
- * @param {string} name - Name of the metric
- * @param {string} help - Help description for the metric
- * @param {Array.<string>} labels - Labels
- * @constructor
- */
-function Counter(name, help, labels) {
-	var config;
-	if(isObject(name)) {
-		config = extend({
-			labelNames: [],
-			registers: [ globalRegistry ]
-		}, name);
-	} else {
+class Counter {
+	 /**
+	 * Counter
+	 * @param {string} name - Name of the metric
+	 * @param {string} help - Help description for the metric
+	 * @param {Array.<string>} labels - Labels
+	 */
+	constructor(name, help, labels) {
+		var config;
+		if(isObject(name)) {
+			config = extend({
+				labelNames: [],
+				registers: [ globalRegistry ]
+			}, name);
+		} else {
 		//Backwards compability - will be depricated
-		config = {
-			name: name,
-			help: help,
-			labelNames: labels,
-			registers: [ globalRegistry ]
-		};
-	}
-
-	if(!config.help) {
-		throw new Error('Missing mandatory help parameter');
-	}
-	if(!config.name) {
-		throw new Error('Missing mandatory name parameter');
-	}
-	if(!validateMetricName(config.name)) {
-		throw new Error('Invalid metric name');
-	}
+			config = {
+				name: name,
+				help: help,
+				labelNames: labels,
+				registers: [ globalRegistry ]
+			};
+		}
 
-	if(!validateLabelNames(config.labelNames)) {
-		throw new Error('Invalid label name');
-	}
+		if(!config.help) {
+			throw new Error('Missing mandatory help parameter');
+		}
+		if(!config.name) {
+			throw new Error('Missing mandatory name parameter');
+		}
+		if(!validateMetricName(config.name)) {
+			throw new Error('Invalid metric name');
+		}
 
-	this.name = name;
-	this.name = config.name;
-	this.hashMap = {};
+		if(!validateLabelNames(config.labelNames)) {
+			throw new Error('Invalid label name');
+		}
 
-	this.labelNames = (config.labelNames || []);
+		this.name = name;
```

Comment:
Could remove the first assignment of name, even though it exists in current version🙂